src/multiConfigs.ts

```diff
@@ -3,40 +3,51 @@ import * as path from "path";
 import * as fs from "fs";
 import { userInfo } from "os";
 import { SERIALIZED_FILE_EXTENSION } from "./codeMarker";
-
-interface ConfigurationEntry {
-    path: string;
-    username: string;
-}
+import { ConfigurationEntry, WorkspaceRootEntry, ConfigTreeEntry, isConfigurationEntry, isWorkspaceRootEntry, configEntryEquals } from "./types";
 
 let lightEyePath: vscode.Uri;
 let darkEyePath: vscode.Uri;
 
-export class MultipleSavedFindingsTree implements vscode.TreeDataProvider<ConfigurationEntry> {
+export class MultipleSavedFindingsTree implements vscode.TreeDataProvider<ConfigTreeEntry> {
     private configurationEntries: ConfigurationEntry[];
-    private workspacePath: string;
-    private activeUsernames: string[];
+    private rootEntries: WorkspaceRootEntry[];
+    private rootPaths: string[];
+    private activeConfigs: ConfigurationEntry[];
     private username: string;
 
-    private _onDidChangeTreeDataEmitter = new vscode.EventEmitter<ConfigurationEntry | undefined | void>();
+    private _onDidChangeTreeDataEmitter = new vscode.EventEmitter<ConfigTreeEntry | undefined | void>();
     readonly onDidChangeTreeData = this._onDidChangeTreeDataEmitter.event;
 
     refresh(): void {
         this._onDidChangeTreeDataEmitter.fire();
     }
 
-    constructor() {
-        this.workspacePath = vscode.workspace.workspaceFolders![0].uri.fsPath;
+    constructor(context: vscode.ExtensionContext) {
+        this.rootPaths = [];
+        this.getWorkspaceRootPaths();
 
         this.configurationEntries = [];
+        this.rootEntries = [];
+        this.activeConfigs = [];
         this.findAndLoadConfigurationFiles();
 
-        this.activeUsernames = [];
+        const listener = (_event: vscode.WorkspaceFoldersChangeEvent) => {
+            this.rootPaths = [];
+            this.getWorkspaceRootPaths();
+            this.configurationEntries = [];
+            this.findAndLoadConfigurationFiles();
+            this.refresh();
+        };
+
+        const disposable = vscode.workspace.onDidChangeWorkspaceFolders(listener);
+        context.subscriptions.push(disposable);
+
+        this.activeConfigs = [];
```

Comment:
`this.activeConfigs` is already set on line 31

Reply:
This is resolved in 9ae4bf7